pkg/integrations/track/tracker.go

```diff
@@ -122,7 +122,11 @@ func (t *Tracker) NotifyExit(hash string, exitCode int) {
 		"cmd_args_name": iCtx.def.CmdChanReq.IntegrationName,
 		"cmd_args_args": fmt.Sprintf("%+v", iCtx.def.CmdChanReq.IntegrationArgs),
 		"cmd_exit_code": exitCode,
-	})
+	}
+	for k, v := range iCtx.def.CmdChanReq.Metadata {
+		ev["cmd_metadata."+k] = v
+	}
+	ds := protocol.NewEventDataset(ts, ev)
 	data := protocol.NewData("tracker.notifyexit", "1", []protocol.Dataset{ds})
 	t.eventEmitter.Send(fwrequest.NewFwRequest(iCtx.def, nil, nil, data))
```

Comment:
Same comment as @brushknight, doesn't look like we do a check for `cmd_metadata.<something>` is correctly emitted here. Apart from that the logic looks sound.